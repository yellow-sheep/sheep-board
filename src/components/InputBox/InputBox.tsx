import * as React from 'react';

interface InputProps {
  text?: string;
  placeholder?: string;
  name?: string;
}

const InputBox: React.SFC<InputProps> = ({
  text = '',
  placeholder = '',
  name,
}) => {
  const [value, setValue] = React.useState('');
  React.useEffect(() => {
    setValue(text);
  }, []);
  return (
    <input
      name={name}
      value={value}
      onChange={e => setValue(e.target.value)}
      type="text"
      placeholder={placeholder}
    />
  );
};

export { InputBox };
