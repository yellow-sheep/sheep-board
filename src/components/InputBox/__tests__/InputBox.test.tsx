import { render, cleanup } from 'react-testing-library';
import * as React from 'react';
import 'jest-dom/extend-expect';
import { InputBox } from '../InputBox';

describe('InputBox', () => {
  afterEach(cleanup);
  test('should render correctly with no props', () => {
    const wrapper = render(<InputBox />);

    expect(wrapper.container).toMatchSnapshot();
  });

  test('should render text when passed as props', () => {
    const wrapper = render(<InputBox text="test text" />);

    expect(wrapper.container).toMatchSnapshot();
  });

  test('should render placeholder text correctly when passed as props', () => {
    const wrapper = render(<InputBox placeholder="enter title..." />);

    expect(
      wrapper.queryByPlaceholderText('enter title...')
    ).toBeInTheDocument();
  });
});
