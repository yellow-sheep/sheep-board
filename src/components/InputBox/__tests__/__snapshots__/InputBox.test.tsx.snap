// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox should render correctly with no props 1`] = `
<div>
  <input
    placeholder=""
    type="text"
    value=""
  />
</div>
`;

exports[`InputBox should render text when passed as props 1`] = `
<div>
  <input
    placeholder=""
    type="text"
    value="test text"
  />
</div>
`;
